server/kit/router.go

```diff
@@ -0,0 +1,182 @@
+package kit
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/julienschmidt/httprouter"
+)
+
+// Router is an interface to wrap different router implementations.
+type Router interface {
+	Handle(method string, path string, handler http.Handler)
+	HandleFunc(method string, path string, handlerFunc func(http.ResponseWriter, *http.Request))
+	ServeHTTP(w http.ResponseWriter, r *http.Request)
+	SetNotFoundHandler(handler http.Handler)
+}
+
+// RouterOption sets optional Router overrides.
+type RouterOption func(Router) Router
+
+// RouterSelect allows users to override the
+// default use of the Gorilla Router.
+// TODO add type and constants for names + docs
+func RouterSelect(name string) RouterOption {
+	return func(_ Router) Router {
```

Comment:
Tiny one: should we use just `func(Rounter) Router` instead of `func(_ Router) Router`?